src/decode.h

```diff
@@ -763,6 +769,9 @@ void CaptureStatsSetup(ThreadVars *tv, CaptureStats *s);
         (p)->dp = 0;                            \
         (p)->proto = 0;                         \
         (p)->recursion_level = 0;               \
+        if ((p)->ReinitPacket) (p)->ReinitPacket((p)); \
```

Comment:
@victorjulien This check is required because it seems that some tests aren't calling https://github.com/OISF/suricata/pull/5492/files#diff-48e701cfe9e2223cedaa68c2dad70955a5b41c803cc1a3feba33d75a18d35533R752.

I have not been able to identify where this happens.